### DISCUSSION TITLE
Row selection: select all table rows when the table is paginated

### DISCUSSION THREAD

**Discussion by luigiPinnaAevo** [Budibase/budibase#8485] [Help] [answered]
Hello,

I enabled "Allow row selection" in my table because I need to allow users to select a single row or all rows for export table data. The selection and the export data work fine when there isn't pagination but if the table is paginated the row selection works only for the items on the first page and on other pages items remain unselected.

How can I solve this issue?

Thank you

**Comment by melohagan** [comment]
Hey @luigiPinnaAevo 

I believe this is intended functionality. The **Export Data** action should only export selected rows. 
You can select all the rows on one pages, navigate to the next page, select all the rows on the next page. In this case the export will include both pages.

**Reply by luigiPinnaAevo** [reply]
Ok thank you. I thought there was a solution to select all rows (also on other pages) with the 'select all' checkbox even if the items are paginated.

**Reply by luigiPinnaAevo** [reply]
Even if I have 10 pages?

**Reply by melohagan** [reply]
Hey @raventechnologiesinc maybe there is a total limit. How many rows did each page have?

**Reply by melohagan** [reply, marked as answer]
I've raised a feature request here: https://github.com/Budibase/budibase/issues/12841
